graphs/graph_manager.py

```
@@ -0,0 +1,210 @@
+        Raises:
+            DiscordError: If message deletion fails
+        """
+        try:
+            async for message in channel.history(limit=limit):
+                if message.author == channel.guild.me:
+                    try:
+                        await message.delete()
+                        await asyncio.sleep(1)  # Rate limiting
+                        logging.info(self.translations["log_deleted_message"])
+                    except discord.Forbidden:
+                        raise DiscordError("Missing permissions to delete messages")
```

Ruff

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)